sklearn/linear_model/glm.py

```diff
@@ -0,0 +1,1813 @@
+    regularizer.
+
+    Minimizes the objective function::
+
+            1/(2s) * deviance(y, h(X*w))
+            + alpha * l1_ratio * ||P1*w||_1
+            + 1/2 * alpha * (1 - l1_ratio) * w*P2*w
+
+    with inverse link function `h` and s=sum of `sample_weight` (which equals
+    n_samples for `sample_weight=None`).
+    For `P1=P2=identity`, the penalty is the elastic net::
+
+            alpha * l1_ratio * ||w||_1
+            + 1/2 * alpha * (1 - l1_ratio) * ||w||_2^2
+
+    If you are interested in controlling the L1 and L2 penalty
+    separately, keep in mind that this is equivalent to::
+
+            a * L1 + b * L2
+
+    where::
+
+            alpha = a + b and l1_ratio = a / (a + b)
+
+    The parameter `l1_ratio` corresponds to alpha in the glmnet R package while
+    'alpha' corresponds to the lambda parameter in glmnet. Specifically,
+    l1_ratio = 1 is the lasso penalty.
+
+    Read more in the :ref:`User Guide <Generalized_linear_regression>`.
+
+    The fit itself does not need Y to be from an EDM, but only assumes
+    the first two moments :math:`E[Y_i]=\\mu_i=h(\\eta_i)` and
+    :math:`Var[Y_i]=\\frac{\\phi}{w_i} v(\\mu_i)`.
+
+    The parameters :math:`w` (`coef_` and `intercept_`) are estimated by
+    (penalized) maximum likelihood which is equivalent to minimizing the
+    deviance.
+
+    For `alpha` > 0, the feature matrix `X` should be standardized in order to
+    penalize features equally strong. Call
+    :class:`sklearn.preprocessing.StandardScaler` before calling ``fit``.
+
+    TODO: Estimation of the dispersion parameter phi.
+
+    If the target `y` is a ratio, appropriate weights `w` should be provided.
+    As an example, consider Poission distributed counts `z` (integers) and
+    weights `w=exposure` (time, money, persons years, ...). Then you fit
+    `y = z/w`, i.e. ``GeneralizedLinearModel(family='poisson').fit(X, y,
+    sample_weight=w)``. The weights are necessary for the right meanself.
+    Consider :math:`\\bar{y} = \\frac{\\sum_i w_i y_i}{\\sum_i w_i}`,
+    in this case one might say that `y` has a 'scaled' Poisson distributions.
+    The same holds for other distributions.
+
+    Parameters
+    ----------
+    alpha : float, optional (default=1)
+        Constant that multiplies the penalty terms und thus determines the
+        regularization strength.
+        See the notes for the exact mathematical meaning of this
+        parameter.``alpha = 0`` is equivalent to unpenalized GLMs. In this
+        case, the design matrix X must have full column rank
+        (no collinearities).
+
+    l1_ratio : float, optional (default=0)
+        The elastic net mixing parameter, with ``0 <= l1_ratio <= 1``. For
+        ``l1_ratio = 0`` the penalty is an L2 penalty. ``For l1_ratio = 1`` it
+        is an L1 penalty.  For ``0 < l1_ratio < 1``, the penalty is a
+        combination of L1 and L2.
+
+    P1 : {None, array-like}, shape (n_features*,), optional \
```

Comment:
I don't think you need this asterisk, or the note below... `n_features` always refers to X.shape[1] when fitting in scikit-learn (unless referring to the output of a transformer)